topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorTest.java

```diff
@@ -141,12 +139,11 @@ private Map<String, String> map(String... pairs) {
     public void testOnKafkaTopicAdded_ignorable(VertxTestContext context) {
         KafkaTopic kafkaTopic = new KafkaTopicBuilder().withMetadata(new ObjectMetaBuilder().withName("non-topic").build()).build();
 
-        Checkpoint async = context.checkpoint();
         K8sTopicWatcher w = new K8sTopicWatcher(topicOperator, Future.succeededFuture(), () -> { });
         w.eventReceived(ADDED, kafkaTopic);
         mockKafka.assertEmpty(context);
         mockTopicStore.assertEmpty(context);
-        async.flag();
+        context.completeNow();
```

Comment:
I think you're correct that we could change this to be a fully synchronous test. I did consider doing that but wondered if it would be better to leave it consistent to reduce future possibilities that people mistakenly copy and paste this test. As I think what happened with the other tests in this class is originally the resourceAdded function was always the last step in the test, but then it got reused in another test where the intent wasn't understood. So that's why I left it as "asynchronous"

Reply:
Seems a reasonable justification to me, let's leave it as is.